File: java/TuduManager/domain/src/main/java/net/seymourpoler/tudumanager/domain/todo/update/UpdateTodoService.java
package net.seymourpoler.tudumanager.domain.todo.update;

import net.seymourpoler.tudumanager.domain.ErrorCodes;
import net.seymourpoler.tudumanager.domain.ServiceExecutionResult;
import net.seymourpoler.tudumanager.domain.todo.delete.IExistTodoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class UpdateTodoService implements  IUpdateTodoService{
    private final IExistTodoRepository existTodoRepository;
    private final IFindTodoByIdRepository findTodoRepository;
    private final IUpdateTodoRepository updateTodoRepository;

    @Autowired
    public UpdateTodoService(
            IExistTodoRepository existTodoRepository,
            IFindTodoByIdRepository findTodoRepository,
            IUpdateTodoRepository updateTodoRepository) {
        this.existTodoRepository = existTodoRepository;
        this.findTodoRepository = findTodoRepository;
        this.updateTodoRepository = updateTodoRepository;
    }

    @Override
    public ServiceExecutionResult update(TodoUpdatingRequest request) {
        if(!existTodoRepository.exist(request.id())){
            var errors = List.of(new net.seymourpoler.tudumanager.domain.Error("id", ErrorCodes.NotFound));
            return ServiceExecutionResult.of(errors);
        }
        if(request.title() == null || request.title().trim().equals("")){
            var errors = List.of(new net.seymourpoler.tudumanager.domain.Error("title", ErrorCodes.Required));
            return ServiceExecutionResult.of(errors);
        }
        final Integer maximumNumberOfCharacters = 250;
        if(request.title().length() > maximumNumberOfCharacters){
            var errors = List.of(new net.seymourpoler.tudumanager.domain.Error("title", ErrorCodes.InvalidLength));
            return ServiceExecutionResult.of(errors);
        }

        var todo = findTodoRepository.find(request.id());
        todo.update(request.title(), request.description());
        updateTodoRepository.update(todo);

        return ServiceExecutionResult.ok();
    }
}

File: dotnet/react.mvc-core.postgres.dapper/TuduManager/src/TuduManayer.Domain/Todo/UpdateTodoService.cs
using System;
using System.Collections.Generic;
using TuduManayer.Domain.Todo.Delete;

namespace TuduManayer.Domain.Todo
{
    public interface IUpdateTodoService
    {
        ServiceExecutionResult Update(TodoUpdatingArgs todoUpdatingArgs);
    }

    public class TodoUpdatingArgs
    {
        public TodoUpdatingArgs(int id, string title, string description)
        {
            Id = id;
            Title = title;
            Description = description;
        }

        public int Id { get; }
        public string Title { get; }
        public string Description { get; }
    }

    public class UpdateTodoService : IUpdateTodoService
    {
        public static int MaximumNumberOfCharacters = 255;
        
        private readonly IExistTodoRepository existTodoRepository;
        
        public UpdateTodoService(IExistTodoRepository existTodoRepository)
        {
            this.existTodoRepository = existTodoRepository;
        }

        public ServiceExecutionResult Update(TodoUpdatingArgs todoUpdatingArgs)
        {
            var errors = new List<Error>();
            if (string.IsNullOrWhiteSpace(todoUpdatingArgs.Title))
                errors.Add(
                    Error.With(nameof(todoUpdatingArgs.Title), ErrorCodes.Required));
            else if (todoUpdatingArgs.Title.Length > MaximumNumberOfCharacters)
            {
                errors.Add(
                    Error.With(nameof(todoUpdatingArgs.Title), ErrorCodes.InvalidLength));
            }
            
            if (!string.IsNullOrEmpty(todoUpdatingArgs.Description) && todoUpdatingArgs.Description.Length > MaximumNumberOfCharacters)
            {
                errors.Add(
                    Error.With(nameof(todoUpdatingArgs.Description), ErrorCodes.InvalidLength));
            }

            if (errors.IsNotEmpty())
            {
                return ServiceExecutionResult.WithErrors(errors);
            }

            if (IsNotExistTodo(todoUpdatingArgs.Id))
            {
                return ServiceExecutionResult.WithErrors(new List<Error> {
                        Error.With(nameof(todoUpdatingArgs.Id), ErrorCodes.NotFound)});
            }
            throw new NotImplementedException();
        }

        private bool IsNotExistTodo(int todoId)
        {
            return !existTodoRepository.Exist(todoId);
        }
    }
}